lib/steps.js

```diff
@@ -730,20 +741,49 @@ After(async function(scenario) {
     let safe_filename = await scope.getSafeScenarioFilename( scope, { prefix: `error` });
     await scope.page.screenshot({ path: `${ safe_filename }.jpg`, type: `jpeg`, fullPage: true });
 
-    // For framework development, if we're testing `unexpected` errors,
-    // only really fail if our special framework-specific test fails.
-    if ( scenario.result.status === `failed` && scope.error_report_id ) {
-      try {
-        await scope.testReport( scope, { stored_report_key: scope.error_report_id });
+    // If we're testing this framework itself for correct reports of
+    // 'unexpected' behavior.
+    let expected_report_text = scope.expected_fail_report || scope.expected_pass_report;
+    // try/catch to let us be sure to reset report values
+    try {
+      if ( scope.expected_fail_report ) {
+        expect( scenario.result.status ).to.equal( `failed` );
         scenario.result.status = `passed`;
-      } catch ( err ) {
-        scope.error_report_id = null;
-        throw err;
+      } else if ( scope.expected_pass_report ) {
+        expect( scenario.result.status ).to.equal( `passed` );
       }
+
+      if ( expected_report_text ) {
+        await scope.testReport2( scope, { expected_report_text });
+      }
+
+    } catch ( err ) {
+      // Reset report values no matter what so they don't mess up future scenarios
+      // Maybe this should be in `Before`, but it would be so out of context there...
+      scope.expected_fail_report = null;
+      scope.expected_pass_report = null;
```

Comment:
Shouldn't these values get reset even if no error got thrown?

Reply:
Good eye. It'll definitely be fixed before this is merged.